src/NServiceBus.Gateway/Gateway.cs

```diff
@@ -94,10 +94,23 @@ static void RegisterChannels(FeatureConfigurationContext context, IManageReceive
             {
                 channelReceiverFactory = s => new ChannelReceiverFactory(typeof(HttpChannelReceiver)).GetReceiver(s);
                 channelSenderFactory = s => new ChannelSenderFactory(typeof(HttpChannelSender)).GetSender(s);
+
+                var repliesDisabled = context.Settings.GetOrDefault<bool>("Gateway.DisableReplyChannelCheck");
+                if (repliesDisabled == false)
+                {
+                    CheckForNonWildcardDefaultChannel(channelManager);
+                }
+                RegisterHttpListenerInstaller(context, channelManager);
             }
+        }
 
-            var enableHttpListener = !usingCustomChannelProviders;
```

Comment:
I think it would be better to set the flag for the installer in some object and resolve that object from the container, like it is done in NH. Various containers might have different behavior when re-registering thins in the container and the installers are first registered by convention in the core.